include/Target/Verilog/VerilogEmitter.h

```diff
@@ -38,8 +46,28 @@ class VerilogEmitter {
   // Globally unique identifiers for values
   int64_t value_count_;
 
+  // A helper to generalize the work of emitting a func.func and a
+  // secret.generic
+  LogicalResult printFunctionLikeOp(Operation *op,
+                                    llvm::StringRef verilogModuleName,
+                                    ArrayRef<BlockArgument> arguments,
+                                    TypeRange resultTypes,
+                                    // FIXME: what is the right type for this?
+                                    // would like to pass something like
+                                    // ArrayRef<Block>
```

Comment:
nit: still relevant? I figure since this matched `getBlocks()` return value this is probably easiest.

Reply:
I was just waiting for a response to my question about it: https://discord.com/channels/636084430946959380/636732535434510338/1175228413070561322

Will fix.